### ISSUE TITLE
Upgrading synapse from 1.37 to 1.38 seems to break BOTS which uses the matrix-nio client library

### ISSUE THREAD

**Issue by toushin-taishi** [matrix-org/synapse#10409]
### Description

BOTS which uses the matrix-nio library stopped working after the upgrade to 1.38.
- matrix-eno-bot
- reminderbot

### Steps to reproduce
1. Upgrade 1.37 to 1.38
2. matrix-eno-bot and reminderbot will stop working

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from _homeserver.log_, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.
-->

### Version information

<!-- IMPORTANT: please answer the following questions, to help us narrow down the problem -->

<!-- Was this issue identified on matrix.org or another homeserver? -->
- **Homeserver**: self-hosted

If not matrix.org:

<!--
 What version of Synapse is running?

You can find the Synapse version with this command:

$ curl http://localhost:8008/_synapse/admin/v1/server_version

(You may need to replace `localhost:8008` if Synapse is not configured to
listen on that port.)
-->
- **Version**: 1.38

- **Install method**: DEB packages
<!-- examples: package manager/git clone/pip  -->

- **Platform**: Baremetal Ubuntu 18.04.5 LTS 
<!--
Tell us about the environment in which your homeserver is operating
distro, hardware, if it's running in a vm/container, etc.
-->

**Comment by dklimpel** [comment]
You have to update matrix-nio.
See: https://matrix.org/blog/posts#dept-of-sdks-and-frameworks-%F0%9F%A7%B0
and: https://github.com/poljar/matrix-nio/blob/master/CHANGELOG.md#0184---2021-07-14

**Comment by toushin-taishi** [comment]
Thank you. Updating matrix-nio fixed the issue with 1.38.
